### ISSUE TITLE
Windows e2e in GCE: secrets tests failing due to unexpected output

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#70882]
CI tests for windows in GCE: https://testgrid.k8s.io/google-windows#windows-prototype
The tests use the repo-list: https://raw.githubusercontent.com/e2e-win/e2e-win-prow-deployment/master/repo-list.txt

Failure message:
```
Nov  9 16:59:05.529: INFO: Waiting for pod pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6 to disappear
Nov  9 16:59:05.532: INFO: Pod pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6 no longer exists
Nov  9 16:59:05.532: INFO: Unexpected error occurred: expected "mode of file \"/etc/secret-volume/data-1\": -rw-r--r--" in container output: Expected
    <string>: mode of Windows file "/etc/secret-volume/data-1": -rwx---r-x
    content of file "/etc/secret-volume/data-1": value-1
    
    
to contain substring
    <string>: mode of file "/etc/secret-volume/data-1": -rw-r--r--
```

The output differs from the expected string with an additional "Windows" string as well as different file permissions.

Related events:
```
Nov  9 16:59:05.535: INFO: At 2018-11-09 16:56:16 +0000 UTC - event for pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6: {default-scheduler } Scheduled: Successfully assigned e2e-tests-secrets-t24tf/pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6 to e2e-86-be9ce-minion-windows-group-mvxb
Nov  9 16:59:05.535: INFO: At 2018-11-09 16:57:23 +0000 UTC - event for pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6: {kubelet e2e-86-be9ce-minion-windows-group-mvxb} Pulled: Container image "e2eteam/mounttest:1.0" already present on machine
Nov  9 16:59:05.535: INFO: At 2018-11-09 16:57:24 +0000 UTC - event for pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6: {kubelet e2e-86-be9ce-minion-windows-group-mvxb} Created: Created container
Nov  9 16:59:05.535: INFO: At 2018-11-09 16:57:34 +0000 UTC - event for pod-secrets-5f43d8aa-e440-11e8-a360-0a580a3c99f6: {kubelet e2e-86-be9ce-minion-windows-group-mvxb} Started: Started container
```

According to @PatrickLang, "Those privileges are more or less the same for a file so I think we should address it as we work to make that test more OS-independent. That test case is relying on a build of busybox to make a best guess at what a unix permission would be based off a Windows ACL. Windows doesn't have an execute permission that could apply to a file"

@kubernetes/sig-windows-misc 
/cc @PatrickLang @adelina-t 
/cc @pjh

**Comment by PatrickLang** [comment]
/assign @bclau 
cc @adelina-t 

I'm seeing this as well on my latest runs. It doesn't seem to matter which Docker version is used - both 17.06 and 18.03 show this behavior.

Claudiu - when you rebuilt busybox to change the environment variable behavior, do you think it changed how it was reporting the permission masks?

**Comment by PatrickLang** [comment]
cc @jsturtevant - this is what I mentioned on Slack

**Comment by PatrickLang** [comment]
Test cases affected:

```
[sig-storage] ConfigMap should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] ConfigMap should be consumable from pods in volume with mappings [NodeConformance] [Conformance]
[sig-storage] Projected configMap should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Projected configMap should be consumable from pods in volume with defaultMode set [NodeConformance] [Conformance]
[sig-storage] Projected configMap should be consumable from pods in volume with mappings [NodeConformance] [Conformance] 
[sig-storage] Projected configMap should be consumable from pods in volume with mappings and Item mode set [NodeConformance] [Conformance]
[sig-storage] Projected downwardAPI should set DefaultMode on files [NodeConformance] [Conformance]
[sig-storage] Projected downwardAPI should set mode on item file [NodeConformance] [Conformance] 
[sig-storage] Projected secret should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Projected secret should be consumable from pods in volume with defaultMode set [NodeConformance] [Conformance]
[sig-storage] Projected secret should be consumable from pods in volume with mappings [NodeConformance] [Conformance]
[sig-storage] Projected secret should be consumable from pods in volume with mappings and Item Mode set [NodeConformance] [Conformance]
[sig-storage] Projected secret should be consumable in multiple volumes in a pod [NodeConformance] [Conformance]
[sig-storage] Secrets should be able to mount in a volume regardless of a different secret existing with same name in different namespace [NodeConformance] [Conformance]
[sig-storage] Secrets should be able to mount in a volume regardless of a different secret existing with same name in different namespace [NodeConformance] [Conformance]
[sig-storage] Secrets should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Secrets should be consumable from pods in volume with mappings [NodeConformance] [Conformance]
[sig-storage] Secrets should be consumable in multiple volumes in a pod [NodeConformance] [Conformance]
```

**Comment by adelina-t** [comment]
Some of these test cases would not work on windows: for example:

( Projected ) configMap / Secret should be consumable from pods in volume with mappings ( and Item Mode) 
( Projected ) configMap / Secret should be consumable from pods in volume with defaultMode set
[sig-storage] Projected downwardAPI should set DefaultMode on files [NodeConformance] [Conformance]
[sig-storage] Projected downwardAPI should set mode on item file [NodeConformance] [Conformance] 

There is no direct mapping between the mode the tests expect on files ( Unix ) and how it works on Windows. These should be skipped. 

The only ones that should work here are: 

[sig-storage] ConfigMap should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Projected configMap should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Projected secret should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Secrets should be able to mount in a volume regardless of a different secret existing with same name in different namespace [NodeConformance] [Conformance]
[sig-storage] Secrets should be consumable from pods in volume [NodeConformance] [Conformance]
[sig-storage] Secrets should be consumable in multiple volumes in a pod [NodeConformance]

cc @bclau  Can you take a look in the images and see what's causing these to fail.

**Comment by bclau** [comment]
Hello.

Hm, 705 permissions? That is a bit strange when "others" have more permissions than the user's group. As far as I know, the typical Windows permissions are 775:

    kubectl exec -n ns1 win-busybox -- stat.exe -c "%a" busybox.exe
    775

Can you also try this command in a Windows busybox container?

And as far as permissions checking goes, it seems that there's a commit missing [1]. Will create a PR with it.

[1] https://github.com/bclau/kubernetes/commit/e901f1bc7b31709fc8eeef6126c27d1ae13b12ca

**Comment by adelina-t** [comment]
PR: https://github.com/kubernetes/kubernetes/pull/71036

**Comment by michmike** [comment]
@adelina-t  if this issue is fixed, can we please close it? thanks!

**Comment by adelina-t** [comment]
This issue has been fixed by pr #71036 .
/close
